### PR TITLE
Support API URL configuration in JS and Python SDKs

### LINKED ISSUE #298: [JS, Python] Support `apiUrl` configuration option
Currently only the .NET, Go, and Java SDKs support the `apiUrl` option that replaces the combination of `apiScheme` and `apiHost`.

We should also extend this to JS and Python, ensuring that we document using this instead of scheme/host and deprecate (?) scheme/host

[ ] JS
[ ] Python

### PR DESCRIPTION
**Pull request by ewanharris**
## Description

Adds support for configuring the API URL in the JS and Python SDKs instead of configuring a scheme + host. This is attempted in a backwards compatible manner, including making types optional.

Also copies across a change from the Python SDK where changes were made to the formatting of code in the README

## References

Closes #298

JS PR: https://github.com/openfga/js-sdk/pull/60
Python PR: https://github.com/openfga/python-sdk/pull/67

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: config/clients/js/template/configuration.mustache [hunk 1]
```diff
@@ -58,18 +65,28 @@ export class Configuration {
    */
   private static sdkVersion = "{{packageVersion}}";
 
+  /**
+   * provide the full api URL
```
Comment (rhamzeh) [inline config/clients/js/template/configuration.mustache:69]: Can you append the example here?

`provide the full api URL (e.g. `https://api.{{sampleApiDomain}}`)`

File: config/clients/python/template/client/configuration.mustache [hunk 2]
```diff
@@ -14,13 +14,14 @@ class ClientConfiguration(Configuration):
             self,
             api_scheme="https",
             api_host=None,
+            api_url=None,
```
Comment (rhamzeh) [inline config/clients/python/template/client/configuration.mustache:17]: Shouldn't `api_url` be at the end for it to not be a breaking change?

Comment (ewanharris) [reply config/clients/python/template/client/configuration.mustache:17]: (caveat here I'm not experience with python)

I thought this, but after looking through the python docs I thought it was safe to include here as we're using keyword arguments not positional arguments. Could totally be wrong though so I'll try and check to determine

Comment (rhamzeh) [reply config/clients/python/template/client/configuration.mustache:17]: Unfortunately you can do this in Python:

```py
configuration = ClientConfiguration('http', 'localhost:8080', 'some-store-id', credentials)
```

Screenshot 2024-02-05 at 8 40 53 AM

What I would suggest is adding it at the end and add a TODO: so we can move it to first when we drop the deprecated fields in the next few major changes

Comment (ewanharris) [reply config/clients/python/template/client/configuration.mustache:17]: Ah shoot I had overlooked that this was allowed, I'll change the PR to no longer break this.

File: config/clients/js/template/configuration.mustache (hunk 1 above)
Comment (ewanharris) [reply config/clients/js/template/configuration.mustache]: Fixed in 9ad45dc73c895d6f47060852bc42994ff3ce025e

File: config/clients/python/template/client/configuration.mustache (hunk 2 above)
Comment (ewanharris) [reply config/clients/python/template/client/configuration.mustache]: Fixed in 87c5fa862915ed2851cd877658fc280ab1e370a4

**Comment by ewanharris** [comment]
Build issue is fixed by #303 303

**Pull request merged**